### PR TITLE
fixup ETag properties and odata metadata

### LINKED ISSUE #14679: Ensure entity PartitionKey and RowKey can be nullable
If partitionKey & rowKey are not explicitly specified in the Select expression for query operations, they are not returned in the result objects. Ensure this is handled properly by the client.

### LINKED ISSUE #15385: TableEntity should not have odata properties
TableEntity objects should have all their odata properties removed or replaced with standard properties.

`odata.metadata` - remove
`odata.etag` - remove and add as `ETag`

### PR DESCRIPTION
**Pull request by christothes**
closes #15385 
closes #14679

### CODE REVIEW

File: sdk/tables/Azure.Data.Tables/src/TableConstants.cs [hunk 1]
```diff
@@ -37,7 +37,8 @@ internal static class PropertyNames
             public const string PartitionKey = "PartitionKey";
             public const string RowKey = "RowKey";
             public const string EtagOdata = "odata.etag";
-            public const string ETag = nameof(ITableEntity.ETag);
+            public const string OdataMetadata = "odata.metadata";
+            public const string ETag = "ETag";
```
Comment (jsquire) [inline sdk/tables/Azure.Data.Tables/src/TableConstants.cs:41]: (extreme) nit:  A bit odd to have it `ETag` here and `EtagOdata` a short bit above.

File: sdk/tables/Azure.Data.Tables/src/TableEntity.cs [hunk 2]
```diff
@@ -58,8 +58,8 @@ public string RowKey
         /// <value>An <see cref="ETag"/> containing the ETag value for the entity.</value>
         public ETag ETag
         {
-            get { return new ETag(GetString(TableConstants.PropertyNames.EtagOdata)); }
-            set { _properties[TableConstants.PropertyNames.EtagOdata] = value.ToString(); }
+            get { return new ETag(GetString(TableConstants.PropertyNames.ETag)); }
+            set { _properties[TableConstants.PropertyNames.ETag] = value.ToString(); }
```
Comment (jsquire) [inline sdk/tables/Azure.Data.Tables/src/TableEntity.cs:62]: Safe to assume that ETag is a struct and cannot be `null` here?

**Review by jsquire — APPROVED** [review]
LGTM

File: sdk/tables/Azure.Data.Tables/src/TableEntity.cs (hunk 2 above)
Comment (christothes) [reply sdk/tables/Azure.Data.Tables/src/TableEntity.cs:62]: correct

File: sdk/tables/Azure.Data.Tables/src/TableConstants.cs (hunk 1 above)
Comment (christothes) [reply sdk/tables/Azure.Data.Tables/src/TableConstants.cs:40]: fixed 😄

**Review by pakrym — APPROVED** [review]

**Pull request merged**